packages/web-components/src/components/locale-modal/__stories__/locale-modal.stories.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2020
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { html } from 'lit-element';
+import inPercy from '@percy-io/in-percy';
+import ifNonNull from 'carbon-custom-elements/es/globals/directives/if-non-null';
+import textNullable from '../../../../.storybook/knob-text-nullable';
+import '../locale-modal-container';
+import localeData from './locale-data.json';
+import styles from './locale-modal.stories.scss';
+import readme from './README.stories.mdx';
+
+export const Default = ({ parameters }) => {
+  const { langDisplay, localeList } = parameters?.props?.['dds-locale-modal-container'];
+  return html`
+    <style>
+      ${styles}
+    </style>
+    <dds-locale-modal-container lang-display="${ifNonNull(langDisplay)}" open .localeList="${ifNonNull(localeList)}">
+    </dds-locale-modal-container>
+  `;
+};
+
+export default {
+  title: 'Components/Locale modal',
+  parameters: {
+    ...readme.parameters,
+    knobs: {
+      'dds-locale-modal-container': ({ groupId }) => ({
+        langDisplay: textNullable('Display language (lang-display)', 'United States — English', groupId),
+      }),
+    },
+    props: {
+      'dds-locale-modal-container': {
+        // Lets `<dds-locale-modal-container>` load the nav links if `CORS_PROXY` is set
+        localeList: process.env.CORS_PROXY && !inPercy() ? undefined : localeData,
```

Comment:
Just wondering if in practice, we should be setting both the localeList and lang-display for the user, and not have this be props? The data is typically driven by the LocaleAPI, and the display name is coming from the JSON data via `getLangDisplay` method in the services package: https://github.com/carbon-design-system/ibm-dotcom-library/blob/a3c7463c140a01fa5ef2066c91f257da5a0c7d59/packages/services/src/services/Locale/Locale.js#L259

Reply:
You are right @jeffchew, in the most typical use cases we set it from our services. The property is for mocking (like `inPercy()` case) and more manual usage.